src/main/java/racingcar/domain/CarRepository.java

```diff
@@ -0,0 +1,70 @@
+package racingcar.domain;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static racingcar.util.RandomUtil.generateRandomNumber;
+
+
+public class CarRepository {
+
+    private static final String DUPLICATED = "[ERROR] 중복된 자동차 이름은 허용되지 않습니다.";
+
+    private Set<Car> cars = new HashSet<>();
+
+    public CarRepository(String[] names) {
+        validateDuplication(names);
+
+        for (String name : names) {
+            cars.add(new Car(name));
+        }
+    }
+
+    public Car getCar(String name) {
+        return cars.stream()
+                .filter(c -> c.getName().equals(name))
+                .collect(Collectors.toList())
+                .get(0);
+    }
+
+    public void moveAllCars() {
+        for (Car car : cars) {
+            car.move(generateRandomNumber());
+        }
+    }
```

Comment:
move라고 해도 충분할 것 같네요!

```suggestion
    public void move() {
        for (Car car : cars) {
            car.move(generateRandomNumber());
        }
    }
```

Reply:
동의합니다 수정 했습니다!